src/modules/qt/typewriter.cpp

```diff
@@ -0,0 +1,736 @@
+        }
+
+        pos = 0;
+        size_t pos2 = 0;
+        std::string s;
+        while (pos2 != substr.npos)
+        {
+            pos2 = substr.find_first_of(" \t\n", pos);
+            if (pos2 != substr.npos)
+            {
+                pos2 = substr.find_first_not_of(" \t\n", pos2);
+                if (pos2 != substr.npos)
+                    s = substr.substr(pos, pos2-pos);
+                else
+                    s = substr.substr(pos, -1);
+            }
+            else
+            {
+                s = substr.substr(pos, -1);
+            }
+
+            insertString(s, frame);
+            frame += n;
+            pos = pos2;
+        }
+        i = i_end;
+        ++i;
+    }
+    else if (c == 'l')   // split by lines
+    {
+        ++i;
+
+        // calculate skip from options
+        ParseOptions po;
+        int ret = parseOptions(line, i, po);
+        if (ret < 0)
+        {
+            return ret;
+        }
+        n = getFrameSkipFromOptions(po);
+        if (n == 0)
+            n = 1;
+
+        c = line[i];
+
+        if (c != rangebeg_char)
+            return -i-1;
+
+        ++i;
+        c = line[i];
+
+        size_t i_end = i;
+        while (true)
+        {
+            // search for range end
+            i_end = line.find_first_of(rangeend_char, i_end);
+
+            // if end of string, error
+            if (i_end == line.npos)
+                return -i_end-1;
+
+            // chack if endrange char is not escaped
+            if (line[i_end-1] != escape_char or line[i_end-2] == escape_char)
+                break;
+
+            ++i_end;
+        }
+
+        std::string substr = line.substr(i, i_end-i);
+
+        size_t pos = 0;
+        while (true)
+        {
+            pos = substr.find_first_of(escape_char, pos);
+            if (pos == substr.npos)
+                break;
+            substr.replace(pos, 1, "");
+        }
+
+        pos = 0;
+        size_t pos2 = 0;
+        std::string s;
+        while (pos2 != substr.npos)
+        {
+            pos2 = substr.find_first_of("\n", pos);
+            if (pos2 != substr.npos)
+            {
+                pos2 = substr.find_first_not_of("\n", pos2);
+                if (pos2 != substr.npos)
+                    s = substr.substr(pos, pos2-pos);
+                else
+                    s = substr.substr(pos, -1);
+            }
+            else
+            {
+                s = substr.substr(pos, -1);
+            }
+
+            insertString(s, frame);
+            frame += n;
+            pos = pos2;
+        }
+        i = i_end;
+        ++i;
+    }
+    else    // error
+    {
+        return -i-1;
+    }
+
+    ++i;
+    return i;
+}
+
+/*======================================================================*
+ XML Parser related stuff
+*======================================================================*/
+
+char* clone_string(const char* string) {
+    char* new_string = new char[strlen(string)+1];
```

Comment:
You should `#include <cstring>` for `strlen()` and `strcpy()`

Reply:
Missing include add.